test/addons/argocd/test

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: The RamenDR authors
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+
+from drenv import kubectl
+from drenv import commands
+
+
+def wait(cluster):
+    print("Waiting application to be healthy")
+    kubectl.wait(
+        "application",
+        "guestbook",
+        "--for=jsonpath={.status.health.status}=Healthy",
+        "--namespace=argocd",
+        "--timeout=300s",
+        context=cluster,
+    )
+
+
+if len(sys.argv) != 4:
+    print(f"Usage: {sys.argv[0]} hub cluster1 cluster2")
+    sys.exit(1)
+
+hub = sys.argv[1]
+
+commands.run(
+    "argocd",
+    "app",
+    "create",
+    "guestbook",
+    "--repo",
+    "https://github.com/argoproj/argocd-example-apps.git",
+    "--path",
+    "guestbook",
+    "--dest-server",
+    "https://kubernetes.default.svc",
+    "--dest-namespace",
+    "argo-test",
+    "--sync-option",
+    "CreateNamespace=true",
+    "--sync-policy",
+    "automated",
+    "--kube-context",
+    hub,
```

Comment:
Other issue - if I run the test with fixed kubeconfig (default namespace for all contexts), the test fails:

```
$ addons/argocd/test hub dr1 dr2
Traceback (most recent call last):
  File "/home/nsoffer/src/ramen/test/addons/argocd/test", line 30, in <module>
    commands.run(
  File "/home/nsoffer/src/ramen/test/drenv/commands.py", line 104, in run
    raise Error(args, error, exitcode=p.returncode, output=output.decode())
drenv.commands.Error: Command failed:
   command: ('argocd', 'app', 'create', 'guestbook', '--repo', 'https://github.com/argoproj/argocd-example-apps.git', '--path', 'guestbook', '--dest-server', 'https://kubernetes.default.svc', '--dest-namespace', 'argo-test', '--sync-option', 'CreateNamespace=true', '--sync-policy', 'automated', '--kube-context', 'hub')
   exitcode: 20
   error:
      time="2024-06-09T20:35:42+03:00" level=fatal msg="error retrieving argocd-cm: configmap \"argocd-cm\" not found"

```

Looks like we are missing some --namespace argument, or maybe this is the bug when argocd is not respecting the namespace? In this case we need to run the test using a tempoary kubeconfig in the same way that works for the start hook.

If we need a temporary kubeconfig in 2 cases, it means that we need a helper in drenv to make this easy to use, something like:

    with drenv.temporary_kubeconfig() as env:
        subprocess.run(cmd, env=env)

Reply:
updated code:
1. "--key=value" style is used.
2. yes, test also need first switch to hub context and argocd ns due to argocd issue of respecting kubecontext and namespace parameter.
3. I wrote a function to get env and put this function in a python file inside argocd dir, since such action is only applicable to argocd, and it is easy for me to implement this function in such a way. Please help check it.

Reply:
> 3. I wrote a function to create a temp kubeconfig file and get env, and put this function in a python file inside argocd dir, since such action is only applicable to argocd, and it is easy for me to implement this function in such a way. Please help check it.

@par97 See #1449 adding a help to create tempoary kubeconfig file and env= argument to commands.run/watch that you can use in this PR.

Reply:
I have updated code based on PR 1449.  So need PR 1449 be merged first.